static/tests/frontend/specs/iconPositionOnHiddenElements.js

```diff
@@ -0,0 +1,204 @@
+    utils.addCommentToLine(lines[0], 'c1', function() {
+      utils.addCommentToLine(lines[1], 'c2', function() {
+        utils.addCommentToLine(lines[2], 'c3', function() {
+          utils.addCommentToLine(lines[3], 'c4', function() {
+            utils.addCommentToLine(lines[4], 'c5', function() {
+              done();
+            });
+          });
+        });
+      });
+    });
+  }
+
+  var checkIfHasCommentIconsOnLine = function(linesWhereCommentIsApplied, lineWhereCommentShouldBe) {
+    var $lineWhereCommentShouldBe = helper.padInner$('div').eq(lineWhereCommentShouldBe);
+    var $lineOfCommentIcon = $lineWhereCommentShouldBe.find('.sceneMark--title > span, heading > span').first();
+    var expectedTop = $lineOfCommentIcon.offset().top + 2; // all icons are +2px down to adjust position
+
+    _.each(linesWhereCommentIsApplied, function(lineWhereCommentIsApplied){
+      var commentId = utils.getCommentIdOfLine(lineWhereCommentIsApplied);
+      var $commentIcon = helper.padOuter$('#commentIcons #icon-' + commentId);
+
+      // check if icon exists
+      expect($commentIcon.length).to.be(1);
+      expect($commentIcon.offset().top).to.be(expectedTop);
+    });
+  }
+
+  before(function(done) {
+    createPad(function(){
+      ep_script_toggle_view_test_helper.utils.setEascMode(['script','scene', 'sequence', 'act', 'episode']);
+      apiUtils.startListeningToApiEvents();
+      createCommentsOnLines(COMMENT_LINES, done);
+    });
+    this.timeout(60000);
+  });
+
+  after(function() {
+    // undo frame resize that was done on before()
+    utils.resetScreenSize();
+  });
+
+  context('when has a comment on a script element', function() {
+    context('and SCRIPT is disable', function() {
+      before(function () {
+        ep_script_toggle_view_test_helper.utils.setEascMode(['scene']);
+      });
+
+      it('does not show its comment icon', function (done) {
+        var targetCommentsIsVisible = isCommentIconsVisible([HEADING]);
+        expect(targetCommentsIsVisible).to.be(false);
+        done();
+      });
+
+      context('and user enables SCRIPT', function() {
+        before(function () {
+          ep_script_toggle_view_test_helper.utils.setEascMode(['script']);
+        });
+
+        it('shows its comment icon', function (done) {
+          var targetCommentsIsVisible = isCommentIconsVisible([HEADING]);
+          expect(targetCommentsIsVisible).to.be(true);
+          done();
+        });
+      });
+    });
+  });
+
+  context('when user hides all elements', function() {
+    before(function () {
+      ep_script_toggle_view_test_helper.utils.setEascMode(['']);
+    });
+
+    it('does not show any comment icon', function (done) {
+      var targetCommentsIsNotVisible = isCommentIconsNotVisible(COMMENT_LINES);
+      expect(targetCommentsIsNotVisible).to.be(true);
+      done();
+    });
+  });
+
+  context('when has comment on a scene mark', function() {
+    context('and it is hidden', function() {
```

Comment:
já que não tem outro `context` para se opor ao `and it is hidden`, dá para simplificar a estrutura desse teste e juntar os dois contextos em um só: `context('when has comment on a hidden scene mark', function() {...`